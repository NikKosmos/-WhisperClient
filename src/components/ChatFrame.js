import React from 'react'
import PropTypes from 'prop-types'

class ChatFrame extends React.Component {

    onSelectChat = () => {
        
        const { chat } = this.props
        
        this.props.onSelectChat(chat)
    }

    renderUsersList = (users) => {
        
        if (users && users.length) {

            return users.map((item, index) => {
                return (
                    <div key = { index }>
                        <p>{ item.name }</p>
                    </div>  
                )
            })
        }
    }

    render() {
        const { chat } = this.props

        return (
            <div className = "selectedItem" onClick = { this.onSelectChat }>
                <div>
                    <p>{ chat.name }</p>
                </div>
                <div>
                    <h3>Chat users:</h3><br />
                        <div>
                            { this.renderUsersList(chat.users) }
                        </div>
                </div>
            </div>
        )
    }
}

ChatFrame.propTypes = {
    chat: PropTypes.exact({
        _id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        users: PropTypes.arrayOf(PropTypes.exact({
            _id: PropTypes.string.isRequired,
            name: PropTypes.string.isRequired,
            email: PropTypes.string.isRequired,
        })).isRequired,
    }),
    onSelectChat: PropTypes.func.isRequired,
}

export { ChatFrame }