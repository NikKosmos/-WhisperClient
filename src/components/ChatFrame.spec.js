import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json';

import { ChatFrame } from './ChatFrame'

describe('ChatFrame', () => {
    const mockChat = {
        _id: '0',
        name: 'captains log',
        users: [{
            _id: '0',
            email: 'nemo@captain.com',
            name: 'captain Nemo',
            }],
    }

    const mockChangeCurrentChat = jest.fn()

    it ('should render correctly', () => {
        const testedComponent = shallow(<ChatFrame 
            onSelectChat = { mockChangeCurrentChat } chat = { mockChat }
        />)

        expect(shallowToJson(testedComponent)).toMatchSnapshot();
      })

})

describe('when push the chat for selection', () => {
    const mockChat = {
        _id: '0',
        name: 'captains log',
        users: [{
            _id: '0',
            email: 'nemo@captain.com',
            name: 'captain Nemo',
            }],
    }
    
    const mockMockChangeCurrentChat = jest.fn()

    const myComponent = shallow(<ChatFrame
        onSelectChat = { mockMockChangeCurrentChat } chat = { mockChat }
    />)

    beforeEach(() => {
        myComponent.find('div').first().simulate('click')
    })

    it('calls the props.onSubmit', () => {
        expect(mockMockChangeCurrentChat).toHaveBeenCalledTimes(1);
    })

})