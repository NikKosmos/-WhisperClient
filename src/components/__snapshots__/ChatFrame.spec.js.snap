// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatFrame should render correctly 1`] = `
<div
  onClick={[Function]}
>
  <div>
    <p>
      captains log
    </p>
  </div>
  <div>
    <h3>
      Chat users:
    </h3>
    <br />
    <div>
      <div
        key="0"
      >
        <p>
          captain Nemo
        </p>
      </div>
    </div>
  </div>
</div>
`;
